src/story.ts

```diff
@@ -332,3 +334,19 @@ export type AnnotatedStoryFn<
 export type StoryAnnotationsOrFn<TFramework extends AnyFramework = AnyFramework, TArgs = Args> =
   | AnnotatedStoryFn<TFramework, TArgs>
   | StoryAnnotations<TFramework, TArgs>;
+
+export type ArgsFromMeta<TFramework extends AnyFramework, Meta> = Meta extends {
+  render?: ArgsStoryFn<TFramework, infer RArgs>;
+  loaders?: (infer Loaders)[];
+  decorators?: (infer Decorators)[];
+}
```

Comment:
Should we have a TS "test" for this? At the very least it would document what it is for :)

Reply:
Hehe, yeah, good point, I have tests for it in this PR:
https://github.com/storybookjs/storybook/pull/19512/files
But maybe good to put a simple test here as well.

I was not really sure where to put this. I could also have put it somewhere in the monorepo. Because StoryObj will now use Meta(OrArgs) as generic (typeof meta), I will need something like this in the public types of every renderer (we use it now only for react and svelte). But because there is also framework-specific inference (to infer the args of the component or specific stuff for action arg inference) I can not put all of StoryObj in this package.

Reply:
added a test now